File: src/ca/qc/bdeb/inf203/model/combatants/Veggie.java
package ca.qc.bdeb.inf203.model.combatants;

import ca.qc.bdeb.inf203.model.Action;
import ca.qc.bdeb.inf203.model.Combattant;
import ca.qc.bdeb.inf203.model.Entite;
import ca.qc.bdeb.inf203.model.Etat;
import ca.qc.bdeb.inf203.model.RepresentationImage;

/**
 * Veggie
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
public class Veggie extends Combattant {

    public Veggie() {
        super();
        initialise();
    }

    @Override
    protected void initialise() {
        super.initialise();

        this.attaque = 1;
        this.gentil = false;
        this.hitbox.height = 80;
        this.hitbox.width = 60;
        this.animationFrameRate = 5;

        this.vie = 20;

        this.nbrImagesAnimation.put(Etat.DEPLACEMENT, 5);
        this.nbrImagesAnimation.put(Etat.ATTAQUE, 7);

        this.vitesseAction.put(Action.DEPLACEMENT, -16f);
        this.vitesseAction.put(Action.ATTAQUE, 0.6f);

        this.sprites.put(Etat.DEPLACEMENT, new RepresentationImage(new String[]{"veggies", "normal", "walking"}));
        this.sprites.put(Etat.ATTAQUE, new RepresentationImage(new String[]{"veggies", "normal", "attack"}));
    }

    @Override
    public Entite action(int nbFois) {
        return null;
    }
}

File: assets/js/model/combatants/veggie.js
/**
 * Veggie
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
function Veggie() {
    Combattant.apply(this);
    this.combattant_init();
    this.init();
};
Veggie.prototype = new Combattant();

Veggie.prototype.init = function() {
    this.attaque = 1;
    this.gentil = false;
    this.hitbox.h = 80;
    this.hitbox.w = 60;
    this.animationFrameRate = 5;
    this.etat = navigator.Etat.DEPLACEMENT;

    this.vie = 20;

    this.nbrImagesAnimation[navigator.Etat.DEPLACEMENT] = 5;
    this.nbrImagesAnimation[navigator.Etat.ATTAQUE] = 7;

    this.vitesseAction[navigator.Action.DEPLACEMENT] = -16;
    this.vitesseAction[navigator.Action.ATTAQUE] = 1;

    this.sprites[navigator.Etat.DEPLACEMENT] = new RepresentationImage(["veggies", "normal", "walking"]);
    this.sprites[navigator.Etat.ATTAQUE] = new RepresentationImage(["veggies", "normal", "attack"]);
};
